src/vm/eventpipe.cpp

```diff
@@ -640,10 +650,6 @@ void EventPipe::WriteEventInternal(EventPipeEvent &event, EventPipeEventPayload
     }
     CONTRACTL_END;
 
-    // We can't proceed if tracing is not initialized.
-    if (!s_tracingInitialized)
```

Comment:
I would leave these checks in place until we can replace it with something that is more rigorously correct. With this check  WriteEvent will fail in a relatively narrow race with Shutdown that deletes event before it is used below. Without this check this method will AV anytime Write is called after Shutdown.

Reply:
We can take care of the pre-existing issues later then.